### PR TITLE
Publisher queue limit

### PR DESCRIPTION
**Pull request by pstutz**

### CODE REVIEW

File: src/main/scala/com/ihealthtechnologies/adda/Adda.scala [hunk 1]
```diff
@@ -42,8 +41,9 @@ import akka.util.Timeout
  * PubSub cycles are possible, but in this case the stream completion propagation does not work.
  */
 class Adda(
-  private[this] val privilegedHandlers: List[Any => Unit] = Nil,
-  implicit val system: ActorSystem = ActorSystem(Adda.defaultSystemName)) extends PubSub {
+    private[this] val privilegedHandlers: List[Any => Unit] = Nil,
+    val maxPublisherQueueSize: Int = 100,
```
Comment (jahangirmohammed) [inline src/main/scala/com/ihealthtechnologies/adda/Adda.scala]: Is it intentional that the default here and the default in constructor parameter of CreatePublisher class are different?

File: src/main/scala/com/ihealthtechnologies/adda/pubsub/Publisher.scala [hunk 2]
```diff
@@ -33,10 +32,10 @@ import akka.stream.actor.MaxInFlightRequestStrategy
  */
 class Publisher(
     val trackCompletion: Boolean,
-    val broadcaster: ActorRef) extends ActorSubscriber with ActorLogging {
+    val broadcaster: ActorRef,
+    val maxQueueSize: Int = 1) extends ActorSubscriber with ActorLogging {
 
   val emptyQueue = Queue.empty[Any]
-  val maxQueueSize = 100
```
Comment (jahangirmohammed) [inline src/main/scala/com/ihealthtechnologies/adda/pubsub/Publisher.scala]: I was planning to put the below require in `publishNext` here:
`​require(longerQueue.size <= maxQueueSize, "queue size should never be greater than publisher queue size passed")​ ? `
Thoughts?

Comment (pstutz) [inline src/main/scala/com/ihealthtechnologies/adda/pubsub/Publisher.scala]: Good idea to verify this. I'd rather check this in a publisher test, so we don't pay for it at runtime.

File: src/main/scala/com/ihealthtechnologies/adda/Adda.scala (hunk 1 above)
Comment (pstutz) [inline src/main/scala/com/ihealthtechnologies/adda/Adda.scala]: Nice find, thanks. I'll remove the other default.

**Pull request merged**